services/py-api/tests/integration_tests/test_participant_routes.py

```diff
@@ -0,0 +1,138 @@
+from src.database.repository.participants_repository import ParticipantsRepository
+import pytest
+from httpx import AsyncClient
+from src.database.repository.teams_repository import TeamsRepository
+
+PARTICIPANT_ENDPOINT_URL = "/api/v3/hackathon/participants"
+
+RANDOM_PARTICIPANT_BODY = {"name": "testtest", "email": "testtest@test.com", "is_admin": False}
+IS_ADMIN_TRUE_BODY = {"name": "testtest", "email": "testtest@test.com", "is_admin": True}
+MISSING_REQ_FIELDS_BODY = {"name": "testtest", "is_admin": False}
+
+ADMIN_PARTICIPANT_BODY = {"name": "testtest", "email": "testtest@test.com", "is_admin": True, "team_name": "testteam"}
+EXISTING_TEAM_NAME_BODY = {"name": "testtest", "email": "testtest1@test.com", "is_admin": True, "team_name": "testteam"}
+EXISTING_EMAIL_BODY = {"name": "testtest", "email": "testtest@test.com", "is_admin": True, "team_name": "testteam1"}
+
+
+@pytest.mark.asyncio(loop_scope="session")
+async def test_create_random_participant(
+    async_client: AsyncClient, participant_repository: ParticipantsRepository
+) -> None:
+    resp = await async_client.post(PARTICIPANT_ENDPOINT_URL, json=RANDOM_PARTICIPANT_BODY)
+
+    # Perform cleanup
+    await participant_repository.delete(resp.json()["participant"]["id"])
```

Comment:
As this is a reoccurring code among the tests, I would suggest extracting it as a fixture. Also, it is easy for the developer to forget about doing the cleanup, that's why this behavior should be encapsulated and abstracted for them in the fixture. Here is how to achieve it:

You could use a factory fixture and do the cleanup after the yield. This fixture could accept the different params we use depending on the test case (different body, for example). In the [docs](https://docs.pytest.org/en/6.2.x/fixture.html#factories-as-fixtures) there is an excellent example:
<img width="893" alt="image" src="https://github.com/user-attachments/assets/0fca6734-091c-4e68-a9ee-ecedcf71ed8f">

Reply:
I agree with you on this one! It definitely needs some optimization like this.

Reply:
Fixed in a10d828544025ee80ad7d36e5033831436a2679a